design/design-doc-rbac-manager.md

```diff
@@ -0,0 +1,706 @@
+    rbac.crossplane.io/base-of-namespace-edit: "true"
+rules:
+- apiGroups: [""]
+  resources: [events]
+  verbs: [get, list, watch]
+- apiGroups: [""]
+  resources: [secrets]
+  verbs: ["*"]
+- apiGroups: [apiextensions.crossplane.io]
+  resources: ["*"]
+  verbs: [get, list, watch]
+---
+# This composite resource role aggregates up to any namespace aligned cluster
+# role that declares compatibility by including the XRD label. The same role
+# also aggregates to the crossplane-edit role.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane:composite:examplecomposites.xr.example.org:edit
+  labels:
+    rbac.crossplane.io/aggregate-to-crossplane-edit: "true"
+    rbac.crossplane.io/aggregate-to-namespace-edit: "true"
+    rbac.crossplane.io/xrd: examplecomposites.xr.example.org
+rules:
+- apiGroups: [xr.example.org]
+  resources:
+  - exampleclaims
+  - exampleclaims/status
+  # Composite resources are cluster scoped. Binding this cluster role at the
+  # namespace scope will not actually grant access to these composite resources.
+  # Including them in this cluster role allows one crossplane:composite role to
+  # aggregate to both the crossplane-ns-example-edit and crossplane-edit roles.
+ of this role will not grant
+  - examplecomposites
+  - examplecomposites/status
+  verbs: ["*"]
+```
+
+The RBAC manager creates a namespace aligned cluster roles for any namespace
+with one or more annotations named `rbac.crossplane.io/composites.example.org`
+(where `composites.example.org` is the name of an XRD) and with the value
+`xrd-enabled`.
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: example
+  annotations:
+    rbac.crossplane.io/composites.example.org: xrd-enabled
+    rbac.crossplane.io/examplecomposites.xr.example.org: xrd-enabled
+```
+
+The above namespace, for example, will result in the creation of the following
+cluster role. Note that the latter two role selectors are derived from the
+annotations of the namespace.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane-ns-example-edit
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-namespace-edit: "true"
+      rbac.crossplane.io/base-of-namespace-edit: "true"
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-namespace-edit: "true"
+      rbac.crossplane.io/xrd: composites.example.org
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-namespace-edit: "true"
+      rbac.crossplane.io/xrd: examplecomposites.xr.example.org
+```
+
+### Provider RBAC Mechanics
+
+To create and bind a cluster role for a `Provider` (package):
+
+1. Find its active `ProviderRevision`.
+1. List all CRDs, filter down to those controlled by the active revision.
+1. Create cluster roles that grant RBAC access to said CRDs.
+1. List all service accounts, filter them similarly. There can be only one.
```

Comment:
I think we will name the SA the same name as the revision, so you may not need to list them all

Reply:
What do you think about naming it `crossplane:provider:7f63e3661` (where `7f63e3661` is the revision name) to match the prefix of the RBAC roles the RBAC manager would generate for it?

Reply:
That sounds fine to me, although we can't include `:` as SA's must adhere to DNS-1123 subdomain naming. We could do `crossplane-provider-7f63e3661` or `crossplane.provider.7f63e3661` though.

Reply:
Ah, fair enough. This was just a minor optimisation so I'm happy to go with whatever makes the most sense to you. Using just the revision name seems fine to me. If we do go for a prefix I have a slight preference for hyphen separators.